### ISSUE TITLE
Add option to remove tags from graphite backend

### ISSUE THREAD

**Issue by rtkmhart** [atlassian/gostatsd#35]
Since graphite doesn't support tags, the normalizeBucketName() method at https://github.com/atlassian/gostatsd/blob/master/backend/backends/graphite/graphite.go#L32 translates metrics from something like

`stats.test.counter` 

to

`stats.test.counter.statsd_source_id.192.168.1.1.statsd_source_id.127.0.0.1`

which really breaks things for us (trying to replace etsy/statsd with gostatsd). I got around it by commented out lines 33-38 of graphite.go, but ideally either tags support should be removed entirely from the graphite backend, or add a flag like "--no-tags" to the startup. I did play with --default-tags but there doesn't appear to be way to turn it off completely. I'd put in a PR for the latter, but I don't have  enough Go experience to figure that out in short order.

**Comment by ash2k** [comment]
Looks like out behaviour is significantly different from what original statsd does. Should make it more configurable and it should mimic the original behaviour of course.

**Comment by ash2k** [comment]
@rtkmhart could you try `better-graphite-support` branch to verify it works for you?

**Comment by rtkmhart** [comment]
Thanks for the quick response! Unfortunately the format the graphite backend sends out is like this

```
stats_counts.mike.test.counter %!f(int64=10) 1465832174
stats.mike.test.counter %!d(float64=0.9985744043244373) 1465832174
```

Setup to test this is as follows (all on the same host) with two config files:
config1.toml:
```
[statsdaemon]
    address = "localhost:9125"
```
config2.toml:
```
[graphite]
    address = "localhost:2003"
```
* Run netcat pretending to be graphite: `nc -lk 2003`
* Run instance with graphite backend: `./gostatsd --backends graphite --config-path ./config2.toml --console-addr ":9126" --flush-interval 10s --verbose --metrics-addr ":9125"`
* Run instance with statsd backend: `./gostatsd --backends "statsdaemon" --config-path ./config1.toml --console-addr ":8126" --flush-interval 10s --verbose`
* Send metrics: `while [ 1 ]; do echo "mike.test.counter:1|c" | nc -uq1 localhost 8125; done`

All output from netcat has the formatting string instead of a proper value.

**Comment by rtkmhart** [comment]
I think the format strings are backwards for the Counters section, this seems to work for me. diff as follows:

```
diff --git a/backend/backends/graphite/graphite.go b/backend/backends/graphite/graphite.go
index 533bd8e..710cae1 100644
--- a/backend/backends/graphite/graphite.go
+++ b/backend/backends/graphite/graphite.go
@@ -119,8 +119,8 @@ func (client *client) preparePayload(metrics *types.MetricMap) *bytes.Buffer {
        if client.legacyNamespace {
                metrics.Counters.Each(func(key, tagsKey string, counter types.Counter) {
                        k := sk(key)
-                       fmt.Fprintf(buf, "stats_counts.%s%s %f %d\n", k, client.globalSuffix, counter.Value, now)
-                       fmt.Fprintf(buf, "%s%s%s %d %d\n", client.counterNamespace, k, client.globalSuffix, counter.PerSecond, now)
+                       fmt.Fprintf(buf, "stats_counts.%s%s %d %d\n", k, client.globalSuffix, counter.Value, now)
+                       fmt.Fprintf(buf, "%s%s%s %f %d\n", client.counterNamespace, k, client.globalSuffix, counter.PerSecond, now)
                })
        } else {
                metrics.Counters.Each(func(key, tagsKey string, counter types.Counter) {
```

**Comment by ash2k** [comment]
@rtkmhart Yeah, it was incorrect. I wrote tests and caught few more issues. Could you try the same branch once again?

**Comment by rtkmhart** [comment]
@ash2k Thanks! My initial testing showed this to work. I'm going to try it in production in a select few places this afternoon.

**Comment by rtkmhart** [comment]
I see what's going on, the timestamp that's sent with the metrics never changes.

**Comment by ash2k** [comment]
@rtkmhart I've pushed a fix, let me know if it works for you. Thanks!

**Comment by rtkmhart** [comment]
@ash2k works perfectly, thank you!

**Comment by ash2k** [comment]
@rtkmhart FYI I've just released 0.14.1 with this issue fixed.
